### DISCUSSION TITLE
How to attach Asset to advance many to many relation and PDF thumbnail is not generated

### DISCUSSION THREAD

**Discussion by ashwinrana** [pimcore/pimcore#14497] [Q&A / Support]
Hello all I am using Pimcore 6.9 Enterprise edition. I am trying to attach the asset to advance many to many relation which is inside localized field. I tried to attach it as normal relation but it is not working, then i realized it requires the ElementMetaData. When it tried to create ElementMetaData I cannot use setOwner it is marked as "internal". So can somebody help me and guide me how can i attach the Pimcore Asset to my Data Object not from drag and drop but from PHP code.

And I am also facing the PDF thumbnail not generating issue. I checked the ghostscript, and tried to override the thumnbail generating function but so far i cannot get the issue fix. Can someone help me in this part also?

**Comment by rushitkhunt1996** [comment]
@ashwinrana  have u setup the Maintenance Cron Job it shud be there for PDF preview, check [here](https://pimcore.com/docs/pimcore/current/Development_Documentation/Getting_Started/Installation.html#page_5-Maintenance-Cron-Job)
